central/cve/node/datastore/datastore_impl.go

```diff
@@ -0,0 +1,201 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/cve/common"
+	"github.com/stackrox/rox/central/cve/node/datastore/internal/index"
+	"github.com/stackrox/rox/central/cve/node/datastore/internal/search"
+	"github.com/stackrox/rox/central/cve/node/datastore/internal/store"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sac"
+	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	vulnRequesterOrApproverSAC = sac.ForResources(
+		sac.ForResource(resources.VulnerabilityManagementRequests),
+		sac.ForResource(resources.VulnerabilityManagementApprovals),
+	)
+
+	accessAllCtx = sac.WithAllAccess(context.Background())
+)
+
+type datastoreImpl struct {
+	storage  store.Store
+	indexer  index.Indexer
+	searcher search.Searcher
+
+	cveSuppressionLock  sync.RWMutex
+	cveSuppressionCache common.CVESuppressionCache
+}
+
+func (ds *datastoreImpl) buildSuppressedCache() error {
+	query := pkgSearch.NewQueryBuilder().AddBools(pkgSearch.CVESuppressed, true).ProtoQuery()
+	suppressedCVEs, err := ds.searcher.SearchRawCVEs(accessAllCtx, query)
+	if err != nil {
+		return errors.Wrap(err, "searching suppress CVEs")
+	}
+
+	ds.cveSuppressionLock.Lock()
+	defer ds.cveSuppressionLock.Unlock()
+	for _, cve := range suppressedCVEs {
+		ds.cveSuppressionCache[cve.GetId()] = common.SuppressionCacheEntry{
```

Comment:
Ooo, are we having this cache by id and the other cache by name for backwards compatibility reasons?

Reply:
Actually, below we are doing it by CVE name. Let's standardize?

Reply:
This is a miss, good catch.